### PR TITLE
[CCE] New data source `opentelekomcloud_cce_addon_template_v3`

### PR DESCRIPTION
**Pull request by anton-sidelnikov**
## Summary of the Pull Request
First attempt to obtain templates info

## PR Checklist

* [x] Refers to: #1975 and #1969
* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.
* [x] Release notes added.

## Acceptance Steps Performed

```
=== RUN   TestAccCCEAddonTemplatesV3DataSource_basic
--- PASS: TestAccCCEAddonTemplatesV3DataSource_basic (64.98s)
PASS

Process finished with the exit code 0
```

### CODE REVIEW

**Review by Aloento — DISMISSED** [review]

**Review by Aloento — DISMISSED** [review]

**Review by Aloento — DISMISSED** [review]

**Review by Aloento — DISMISSED** [review]

File: opentelekomcloud/services/cce/data_source_opentelekomcloud_cce_addon_template_v3.go [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+				Optional: true,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceCceAddonTemplatesV3Read(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*cfg.Config)
+	client, err := config.CceV3AddonClient(config.GetRegion(d))
```
Comment (artem-lifshits) [inline opentelekomcloud/services/cce/data_source_opentelekomcloud_cce_addon_template_v3.go:61]: Use client from ctx please

File: opentelekomcloud/services/cce/data_source_opentelekomcloud_cce_addon_template_v3.go [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+	}
+
+	addonTemplates, err := addons.GetTemplates(client).Extract()
+	if err != nil {
+		return fmterr.Errorf("unable to retrieve templates: %s", err)
+	}
+
+	var template addons.AddonTemplate
+	for _, addon := range addonTemplates.Items {
+		if addon.Metadata.Name == d.Get("addon_name") {
```
Comment (artem-lifshits) [inline opentelekomcloud/services/cce/data_source_opentelekomcloud_cce_addon_template_v3.go]: Wouldn't it better to create additional d.Get("addon_name") var and don't call method on each loop run?

File: opentelekomcloud/services/cce/data_source_opentelekomcloud_cce_addon_template_v3.go [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+			template = addon
```
Comment (artem-lifshits) [inline opentelekomcloud/services/cce/data_source_opentelekomcloud_cce_addon_template_v3.go:74]: Shouldn't loop stop when name is found? Or can there be multiple addons with same name? Either way it doesn't look good to me

File: opentelekomcloud/services/cce/data_source_opentelekomcloud_cce_addon_template_v3.go [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+	}
+	if template.Metadata.Id == "" {
+		return fmterr.Errorf("your query returned no results by provided addon name." +
+			" Please change your search criteria and try again")
+	}
+
+	var result addons.Version
+	for _, version := range template.Spec.Versions {
+		if version.Version == d.Get("addon_version") {
+			result = version
```
Comment (artem-lifshits) [inline opentelekomcloud/services/cce/data_source_opentelekomcloud_cce_addon_template_v3.go:87]: Same here

**Review by artem-lifshits — CHANGES_REQUESTED** [review]

File: opentelekomcloud/services/cce/data_source_opentelekomcloud_cce_addon_template_v3.go (hunk 3 above)
Comment (anton-sidelnikov) [reply opentelekomcloud/services/cce/data_source_opentelekomcloud_cce_addon_template_v3.go:74]: only one

File: opentelekomcloud/services/cce/data_source_opentelekomcloud_cce_addon_template_v3.go (hunk 4 above)
Comment (anton-sidelnikov) [reply opentelekomcloud/services/cce/data_source_opentelekomcloud_cce_addon_template_v3.go:87]: info only for one addon

File: opentelekomcloud/services/cce/data_source_opentelekomcloud_cce_addon_template_v3.go (hunk 1 above)
Comment (anton-sidelnikov) [reply opentelekomcloud/services/cce/data_source_opentelekomcloud_cce_addon_template_v3.go:61]: not needed here, it calls only once

File: opentelekomcloud/services/cce/data_source_opentelekomcloud_cce_addon_template_v3.go (hunk 2 above)
Comment (anton-sidelnikov) [reply opentelekomcloud/services/cce/data_source_opentelekomcloud_cce_addon_template_v3.go]: good point

**Review by artem-lifshits — DISMISSED** [review]

**Review by artem-lifshits — APPROVED** [review]

**Review by Aloento — APPROVED** [review]

**Pull request merged**